.github/workflows/test_pysteps.yml

```diff
@@ -0,0 +1,108 @@
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ "ubuntu-latest"] #, "macos-latest", "windows-latest" ]
+        python-version: [ "3.6"] #, "3.8"]
+      max-parallel: 6
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2.0.0
+        with:
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+          channel-priority: flexible
+          activate-environment: test-environment
+
+      - name: Install dependencies
+        env:
+          PACKAGES: ${{env.MINIMAL_DEPENDENCIES}} ${{env.OPTIONAL_DEPENDENCIES}} ${{env.TEST_DEPENDENCIES}}
+          PYSTEPS_DATA_PATH: ${{github.workspace}}/pysteps_data
+        run: |
+          conda install --quiet ${{env.PACKAGES}}
+
+      - name: Install pysteps for MacOS
+        if: matrix.os == 'macos-latest'
+        working-directory: ${{github.workspace}}
+        env:
+          CC: gcc-9
+          CXX: g++-9
+          CXX1X: g++-9
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+        run: |
+          brew update-reset
+          brew update
+          brew install gcc@9
+          gcc-9 --version
+          pip install .
+
+      - name: Install pygrib (not win)
+        if: matrix.os != 'windows-latest'
+        run: conda install --quiet pygrib
+
+      - name: Install pysteps
+        if: matrix.os != 'macos-latest'
+        working-directory: ${{github.workspace}}
+        run: pip install .
+
+      - name: Download pysteps data
+        env:
+          PYSTEPS_DATA_PATH: ${{github.workspace}}/pysteps_data
+        working-directory: ${{github.workspace}}/ci
+        run: python fetch_pysteps_data.py
+
+      - name: Check imports
+        working-directory: ${{github.workspace}}/pysteps_data
+        run: |
+          python -c "import pysteps; print(pysteps.__file__)"
+          python -c "from pysteps import motion"
+          python -c "from pysteps.motion import vet"
+          python -c "from pysteps.motion import proesmans"
+
+      - name: Run tests and coverage report
+        working-directory: ${{github.workspace}}/pysteps_data
+        env:
+          PYSTEPSRC: ${{github.workspace}}/pysteps_data/pystepsrc
+        run: pytest --pyargs pysteps --cov=pysteps --cov-report=xml --cov-report=term -ra
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        env:
+          OS: ${{ matrix.os }}
+          PYTHON: ${{ matrix.python-version }}
+        with:
+          file: coverage.xml
+          directory: ${{github.workspace}}/pysteps_data
+          flags: unit_tests
+          env_vars: OS,PYTHON
+          fail_ci_if_error: true
+          verbose: true
```

Comment:
Github suggests here to have an empty newline to finalize the script. I don't know if it's necessary (not per se for Python at least).